xml/xen_config_options.xml

```diff
@@ -1049,4 +1049,58 @@ alice                             4     1     3   -b-       2.8 2-3</screen>
 <screen>extra_guest_irqs=,512</screen>
   </sect2>
  </sect1>
+ <sect1 xml:id="sec-xen-config-scheduling">
+  <title>Virtual CPU Scheduling</title>
+  <para>
+   &xen; hypervisor schedules virtual CPUs individually across physical CPUs.
+   On modern CPUs that support multiple threads on one core, virtual CPUs can
+   run on the same core in different threads and thus influence each other.
+   Such virtual CPUs can observe performance variations resulting from the
+   actions of a virtual CPU running on the other thread.  When these virtual
+   CPUs belong to different guest systems, these guests can influence each
+   other. The effect can vary, from variations in the guest CPU time accounting
+   to worse scenarios such as <emphasis>side channel attack</emphasis>.
+  </para>
+  <para>
+   <emphasis>Scheduling granularity</emphasis> addresses this problem. You can
+   specify it at boot time by using a &xen; boot parameter:
+  </para>
+  <screen>sched-gran=<replaceable>GRANULARITY</replaceable></screen>
+  <para>
+   Replace <replaceable>GRANULARITY</replaceable> with one of:
+  </para>
+  <variablelist>
+   <varlistentry>
+    <term>cpu</term>
+    <listitem>
+     <para>
+      The regular scheduling of the &xen; hypervisor. Virtual CPUs of different
+      guests can share one physical CPU core. This is the default.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>core</term>
+    <listitem>
+     <para>
+      Virtual CPUs of a virtual core are always scheduled together on one
+      physical core. It never happens that two or more virtual CPUs from
+      different virtual cores are scheduled on the same physical core.
+      Therefore, some physical cores may have some of their CPUs left idle,
+      even if there are virtual CPUs wanting to run. This may slightly impact
+      performance, but it is important to achieve the improved isolation and
+      consistency.
```

Comment:
About the last sentence. I think "slightly" might be a bit vague, or even incorrect, depending on the actual circumstances. In fact, the impact on the performance depends on the benchmark or workload. I'd propose something along the lines of:

"The impact on performance will depend on the actual workload being run inside of the guest systems. In most of the analyzed cases, the observed performance degradation, especially if under considerable load, was smaller than disabling HyperThreading (and hence leaving all the cores with just one thread)."

Reply:
And if we go for (something like) what I proposed, I'd say we could even add a mention/reference to the 'smt' boot parameter, i.e.: https://xenbits.xen.org/docs/unstable/misc/xen-command-line.html#smt-x86

But do we have it documented? I quickly checked, but have not seen it...